StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1205CodeFixProvider.cs

```diff
@@ -0,0 +1,81 @@
+﻿namespace StyleCop.Analyzers.OrderingRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Extensions;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    /// <summary>
+    /// Implements code fixes for <see cref="SA1205PartialElementsMustDeclareAccess"/>.
+    /// </summary>
+    [ExportCodeFixProvider(nameof(SA1205CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1205CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1205PartialElementsMustDeclareAccess.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create("Add public access modifier", token => GetTransformedDocumentAsync(context.Document, diagnostic, SyntaxKind.PublicKeyword, token)), diagnostic);
```

Comment:
:exclamation: The symbol will already have a correct access modifier (if no other fragment has a modifier, the default is `internal`). You'll need to get the declared symbol for the class and determine the effective access modifier from that symbol.